source/MRMesh/MRMeshMetrics.cpp

```diff
@@ -270,29 +270,29 @@ FillHoleMetric getMinTriAngleMetric( const Mesh& mesh )
 }
 
 // Very simple metric for cut mesh holes, only penalty for normal flip and area
-FillHoleMetric getSimplePlanarMetric( const Mesh& mesh, EdgeId e0 )
+FillHoleMetric getSimpleAreaMetric( const Mesh& mesh, EdgeId e0 )
 {
     assert( !mesh.topology.left( e0 ) );
     auto norm = Vector3d();
     for ( auto e : leftRing( mesh.topology, e0 ) )
     {
         norm += cross( Vector3d( mesh.orgPnt( e ) ), Vector3d( mesh.destPnt( e ) ) );
     }
-    norm = norm.normalized();
-
+    auto holeArea = norm.length();
+    auto aspectDenom = 1e-2 / sqrt( std::numeric_limits<double>::max() );
     FillHoleMetric metric;
-    metric.triangleMetric = [&mesh, norm] ( VertId a, VertId b, VertId c )
+    metric.triangleMetric = [&mesh, holeArea, aspectDenom] ( VertId a, VertId b, VertId c )
     {
         Vector3d aP = Vector3d( mesh.points[a] );
         Vector3d bP = Vector3d( mesh.points[b] );
         Vector3d cP = Vector3d( mesh.points[c] );
 
         auto faceNorm = cross( bP - aP, cP - aP );
         auto faceDblArea = faceNorm.length();
-        if ( dot( norm, faceNorm ) < 0.5f * faceDblArea )
-            return BadTriangulationMetric; // DBL_MAX break any triangulation, just return big value to allow some bad meshes
+        auto areaRatio = 0.5f * faceDblArea / holeArea; // [0;1]
```

Comment:
why big triangle area is bad?

Reply:
this metric minizes summary area of new triangles, for planar holes, only reversed cases have bigger area (as far as any triangulation that does not have reversed triangles has same summary area)